### PR TITLE
Make description optional

### LINKED ISSUE #117: Make Description optional

### PR DESCRIPTION
**Pull request by lywich**
Description is now optional.
Default Description is "No Description".
Tasks with default description will not show up when `find` is called.

Closes #117, #122, #128

### CODE REVIEW

File: src/main/java/seedu/task/storage/JsonAdaptedTask.java [hunk 1]
```diff
@@ -77,6 +80,7 @@ public JsonAdaptedTask(@JsonProperty("name") String name,
     public JsonAdaptedTask(Task source) {
         name = source.getName().fullName;
         description = source.getDescription().value;
+        hasDescription = String.valueOf(source.getDescription().getHasDescription());
```
Comment (joyngjr) [inline src/main/java/seedu/task/storage/JsonAdaptedTask.java:83]: Would it be better to abstract out this method as per law of demeter?

File: src/main/java/seedu/task/model/task/DescContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -16,7 +16,8 @@ public DescContainsKeywordsPredicate(String keyphrase) {
 
     @Override
     public boolean test(Task task) {
-        return StringUtil.containsIgnoreCase(task.getDescription().value, keyphrase);
+        return StringUtil.containsIgnoreCase(task.getDescription().value, keyphrase)
+                && task.getDescription().getHasDescription();
```
Comment (joyngjr) [inline src/main/java/seedu/task/model/task/DescContainsKeywordsPredicate.java:20]: Would it be good to allow the user to search for tasks that have no descriptions too?

Comment (joyngjr) [inline src/main/java/seedu/task/model/task/DescContainsKeywordsPredicate.java:20]: How about abstracting this into a method within Description?

File: src/main/java/seedu/task/model/task/Description.java [hunk 3]
```diff
@@ -12,6 +12,15 @@ public class Description {
     public static final String MESSAGE_CONSTRAINTS =
             "Description must contain at least one character";
     public final String value;
+    public final String defaultValue = "No Description";
```
Comment (joyngjr) [inline src/main/java/seedu/task/model/task/Description.java:15]: Would it be better to make it a class variable?

**Review by joyngjr — COMMENTED** [review]
LGTM! Left some comments for your consideration.

File: src/main/java/seedu/task/model/task/DescContainsKeywordsPredicate.java (hunk 2 above)
Comment (lywich) [reply src/main/java/seedu/task/model/task/DescContainsKeywordsPredicate.java:20]: Good for future consideration, will implement if time permits.

File: src/main/java/seedu/task/storage/JsonAdaptedTask.java (hunk 1 above)
Comment (lywich) [reply src/main/java/seedu/task/storage/JsonAdaptedTask.java:83]: Updated as suggested.

File: src/main/java/seedu/task/model/task/DescContainsKeywordsPredicate.java (hunk 2 above)
Comment (lywich) [reply src/main/java/seedu/task/model/task/DescContainsKeywordsPredicate.java:20]: DescContainsKeywordsPredicate is itself an abstraction away from Description hence I think further abstraction would not do much good here.

File: src/main/java/seedu/task/model/task/Description.java (hunk 3 above)
Comment (lywich) [reply src/main/java/seedu/task/model/task/Description.java:15]: Updated as suggested.

**Review by joyngjr — APPROVED** [review]
LGTM!

**Pull request merged**